include/psp2/pss.h

```diff
@@ -13,24 +13,560 @@
+#define PSS_NET_RESOLVER_ESERVERREFUSED            232
+#define PSS_NET_RESOLVER_ENORECORD                 233
+
+#define PSS_NET_SHUT_RD             0
+#define PSS_NET_SHUT_WR             1
+#define PSS_NET_SHUT_RDWR           2
+
+#define PSS_NET_CTL_INFO_CNF_NAME          1
+#define PSS_NET_CTL_INFO_DEVICE            2
+#define PSS_NET_CTL_INFO_ETHER_ADDR        3
+#define PSS_NET_CTL_INFO_MTU               4
+#define PSS_NET_CTL_INFO_LINK              5
+#define PSS_NET_CTL_INFO_BSSID             6
+#define PSS_NET_CTL_INFO_SSID              7
+#define PSS_NET_CTL_INFO_WIFI_SECURITY     8
+#define PSS_NET_CTL_INFO_RSSI_DBM          9
+#define PSS_NET_CTL_INFO_RSSI_PERCENTAGE   10
+#define PSS_NET_CTL_INFO_CHANNEL           11
+#define PSS_NET_CTL_INFO_IP_CONFIG         12
+#define PSS_NET_CTL_INFO_DHCP_HOSTNAME     13
+#define PSS_NET_CTL_INFO_PPPOE_AUTH_NAME   14
+#define PSS_NET_CTL_INFO_IP_ADDRESS        15
+#define PSS_NET_CTL_INFO_NETMASK           16
+#define PSS_NET_CTL_INFO_DEFAULT_ROUTE     17
+#define PSS_NET_CTL_INFO_PRIMARY_DNS       18
+#define PSS_NET_CTL_INFO_SECONDARY_DNS     19
+#define PSS_NET_CTL_INFO_HTTP_PROXY_CONFIG 20
+#define PSS_NET_CTL_INFO_HTTP_PROXY_SERVER 21
+#define PSS_NET_CTL_INFO_HTTP_PROXY_PORT   22
+#define PSS_NET_CTL_INFO_RESERVED1         23
+#define PSS_NET_CTL_INFO_RESERVED2         24
+
+#define PSS_NET_EPOLLIN         0x00000001
+#define PSS_NET_EPOLLOUT        0x00000002
+#define PSS_NET_EPOLLERR        0x00000008
+#define PSS_NET_EPOLLHUP        0x00000010
+#define PSS_NET_EPOLLDESCID     0x00010000
+
+#define PSS_NET_EPOLL_CTL_ADD   1
+#define PSS_NET_EPOLL_CTL_MOD   2
+#define PSS_NET_EPOLL_CTL_DEL   3
+
+typedef struct PssIRegisters {
+	unsigned int size;
+	uint32_t cpsr;
+	uint32_t reg[16];
+	uint32_t tpidrurw;
+	uint32_t teehbr;
+	uint32_t stackBottom;
+	uint32_t stackTop;
+} PSSIntegerRegisters;
+
+typedef struct PssFRegisters {
+	unsigned int size;
+	uint32_t fpscr;
+	struct {
+		uint32_t word[4];
+	} reg[16];
+} PSSFloatRegisters;
+
+typedef struct pss_ucontext {
+	PSSIntegerRegisters iregs;
+	PSSFloatRegisters fregs;
+} pss_ucontext;
+
+typedef struct _PssCryptoContext {
+	int handle;
+	int valid;
+	int size;
+	int type;
+} PssCryptoContext;
+
+typedef struct _PssTimeval {
+	long tv_sec;
+	long tv_usec;
+} PssTimeval;
+
+typedef struct _PssTimespec {
+	int tv_sec;
+	int tv_nsec;
+} PssTimespec;
+
+typedef struct _PssTimezone {
+	int tz_minuteswest;
+	int tz_dsttime;
+} PssTimezone;
+
+typedef struct _PssDateTime {
+    unsigned short year;
+    unsigned short month;
+    unsigned short day;
+    unsigned short hour;
+    unsigned short minute;
+    unsigned short second;
+    unsigned int   microsecond;
+} PssDateTime;
+
+
+typedef struct _PssIoStat {
+	int st_mode;
+	unsigned int st_attr;
+	int64_t st_size;
+	PssDateTime st_ctime;
+	PssDateTime st_atime;
+	PssDateTime st_mtime;
+	unsigned int st_private[6];
+} PssIoStat;
+
+typedef struct _PssNetHostent {
```

Comment:
most type & enum are same as toolchain's or another headers.
so imo good option is
```
#include <netdb.h>
typedef struct hostent PssNetHostent;
```
but i'm not sure we can do possible cross ref newlib's header in here.

Reply:
hm, would have to look through the entire SDK and find where any match?

Reply:
i'm not sure do we need to define twice?5